pallets/data-preservers/src/lib.rs

```diff
@@ -550,22 +613,132 @@ pub mod pallet {
 
             Ok(().into())
         }
+
+        #[pallet::call_index(7)]
+        #[pallet::weight(0)]
+        pub fn start_assignment(
+            origin: OriginFor<T>,
+            profile_id: T::ProfileId,
+            para_id: ParaId,
+            assigner_param: AssignerParameterOf<T>,
+        ) -> DispatchResultWithPostInfo {
+            let assigner = T::AssignmentOrigin::ensure_origin(origin, &para_id)?;
+
+            let mut profile = Profiles::<T>::get(profile_id).ok_or(Error::<T>::UnknownProfileId)?;
+
+            if profile.assignment.is_some() {
+                Err(Error::<T>::ProfileAlreadyAssigned)?
+            }
+
+            if !profile.profile.para_ids.can_assign(&para_id) {
+                Err(Error::<T>::ProfileIsNotElligibleForParaId)?
+            }
+
+            // Add profile id to BoundedVec early in case bound is reached
+            {
+                let mut assignments = Assignments::<T>::get(para_id);
+                let Err(position) = assignments.binary_search(&profile_id) else {
+                    Err(Error::<T>::ProfileAlreadyAssigned)?
+                };
+
+                assignments
+                    .try_insert(position, profile_id)
+                    .map_err(|_| Error::<T>::MaxAssignmentsPerParaIdReached)?;
+
+                Assignments::<T>::insert(para_id, assignments);
+            }
+
+            // TODO: Check Ethereum RPC?
+
+            let witness = T::AssignmentPayment::try_start_assignment(
+                assigner,
+                profile.account.clone(),
+                &profile.profile.assignment_request,
+                assigner_param,
+            )?;
+
+            profile.assignment = Some((para_id, witness));
+            Profiles::<T>::insert(profile_id, profile);
+
+            Self::deposit_event(Event::AssignmentStarted {
+                profile_id,
+                para_id,
+            });
+
+            Ok(().into())
+        }
+
+        #[pallet::call_index(8)]
+        #[pallet::weight(0)]
+        pub fn stop_assignment(
+            origin: OriginFor<T>,
+            profile_id: T::ProfileId,
+            para_id: ParaId,
+        ) -> DispatchResultWithPostInfo {
+            let assigner = T::AssignmentOrigin::ensure_origin(origin, &para_id)?;
```

Comment:
So the container chain manager is the one who can start and stop an assignment, how can the bootnode provider stop an assignment? I guess deleting their profile? Or closing the payment stream?

Reply:
Yes the profile owner should be able to do this. Also:

- on deassignation, we dont pay the profile-id for now (we will see if this is done in the future, but let's keep things simple). If you want to be paid prior to assignation, then call the extrinsic

- Also, yes let's disallow deleting any assigned profiles